RestTemplate component conflict

There is an issue when using this library in existing applications which was not caught by integration testing.

Caused by: org.springframework.beans.factory.support.BeanDefinitionOverrideException: Invalid bean definition with name 'restTemplate' defined in class path resource [com/digitalsanctuary/cf/turnstile/TurnstileConfiguration.class]: Cannot register bean definition [Root bean: class [null]; scope=; abstract=false; lazyInit=null; autowireMode=3; dependencyCheck=0; autowireCandidate=true; primary=false; factoryBeanName=com.digitalsanctuary.cf.turnstile.TurnstileConfiguration; factoryMethodName=restTemplate; initMethodNames=null; destroyMethodNames=[(inferred)]; defined in class path resource [com/digitalsanctuary/cf/turnstile/TurnstileConfiguration.class]] for bean 'restTemplate' since there is already [Root bean: class [null]; scope=; abstract=false; lazyInit=null; autowireMode=3; dependencyCheck=0; autowireCandidate=true; primary=false; factoryBeanName=applicationConfig; factoryMethodName=restTemplate; initMethodNames=null; destroyMethodNames=[(inferred)]; defined in class path resource [com/XXXXXX/util/ApplicationConfig.class]] bound.

Will need to remove that Bean definition and ensure it's setup correctly in the test context I think.

--- a/README.md
+++ b/README.md
@@ -36,7 +36,7 @@ Add the following dependency to your `pom.xml`:
 <dependency>
     <groupId>com.digitalsanctuary</groupId>
     <artifactId>spring-cf-turnstile</artifactId>
-    <version>1.1.1</version>
+    <version>1.1.2</version>
 </dependency>
 ```
 
@@ -46,7 +46,7 @@ Add the following dependency to your `build.gradle`:
 
 ```groovy
 dependencies {
-    implementation 'com.digitalsanctuary:spring-cf-turnstile:1.1.1'
+    implementation 'com.digitalsanctuary:spring-cf-turnstile:1.1.2'
 }
 ```
 

--- a/build.gradle
+++ b/build.gradle
@@ -13,7 +13,7 @@ import com.vanniktech.maven.publish.JavaLibrary
 import com.vanniktech.maven.publish.JavadocJar
 
 group 'com.digitalsanctuary.cf.turnstile'
-version '1.1.1'
+version '1.1.2'
 description = 'SpringBoot Cloudflare Turnstile Library'
 
 ext {

--- a/src/main/java/com/digitalsanctuary/cf/turnstile/TurnstileConfiguration.java
+++ b/src/main/java/com/digitalsanctuary/cf/turnstile/TurnstileConfiguration.java
@@ -24,7 +24,7 @@ import lombok.extern.slf4j.Slf4j;
 public class TurnstileConfiguration {
 
     @Bean
-    public RestTemplate restTemplate(RestTemplateBuilder builder) {
+    public RestTemplate turnstileRestTemplate(RestTemplateBuilder builder) {
         return builder.build();
     }
 

--- a/src/main/java/com/digitalsanctuary/cf/turnstile/TurnstileValidationService.java
+++ b/src/main/java/com/digitalsanctuary/cf/turnstile/TurnstileValidationService.java
@@ -2,6 +2,7 @@ package com.digitalsanctuary.cf.turnstile;
 
 import java.util.HashMap;
 import java.util.Map;
+import org.springframework.beans.factory.annotation.Qualifier;
 import org.springframework.beans.factory.annotation.Value;
 import org.springframework.http.HttpEntity;
 import org.springframework.http.HttpHeaders;
@@ -37,7 +38,7 @@ public class TurnstileValidationService {
      *
      * @param restTemplate the RestTemplate to be used for making HTTP requests.
      */
-    public TurnstileValidationService(RestTemplate restTemplate) {
+    public TurnstileValidationService(@Qualifier("turnstileRestTemplate") RestTemplate restTemplate) {
         this.restTemplate = restTemplate;
     }
 

